src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -1087,6 +1100,20 @@ class ArnoldAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 				ParameterAlgo::setParameter( node, it->first.c_str(), it->second.get() );
 			}
 
+			// Early out for IECoreScene::Procedurals. Arnold's inheritance rules for procedurals are back
+			// to front, with any explicitly set parameters on the procedural node overriding parameters of child
+			// nodes completely. We emulate the inheritance we want in ArnoldProceduralRenderer.
+
+			if( isConvertedProcedural( geometry ) )
+			{
+				// Arnold neither inherits nor overrides visibility parameters. Instead
+				// it does a bitwise `&` between the procedural and its children. Apply
+				// full visibility to the `ginstance` to override the full invisibility
+				// of the procedural it references.
```

Comment:
I think this makes sense to me, but the comment is confusing?  I'm not sure what "override the full invisibility of the procedural it references" means.  I thought it was just "Apply full visibility to the ginstance so that the procedural's visibility settings can apply"

Reply:
Hopefully clarified in https://github.com/GafferHQ/gaffer/pull/3960/commits/5c1e2d91c5ddb83411f08a0cdc9b4c26bc812dba.